### ISSUE TITLE
Vinyl Control not engaging

### ISSUE THREAD

**Issue by djquartz** [mixxxdj/mixxx#13974]
### Bug Description

I setup Mixxx on my Dell 5290 running Windows 10 and found that the Vinyl Control is not engaging with my Rane One setup.

I tried it with 2.4.2, 2.4.5 and 2.4.6_alpha and they all do the same thing.

I double checked the turntable input volume and adjusted it accordingly.

There is a good noisemap signal showing on the scopes. When I enable passthrough I can hear a clean noisemap signal from the vinyl playing.

What happens is when I click the Vinyl Contol button it plays the track for a a seconds in internal mode and then stops.

The vinyl control is not responding to the control vinyl at all even though there is a signal coming through.

When I tested on MacOS, when I clicked the Vinyl Control button it would immediately enable vinyl control and would work.

### Version

2.4.2 - 2.6-alpha

### OS

Windows 10

**Comment by Swiftb0y** [comment]
Hey there, thank you for filling your issue. In order to get to the bottom of this. We would like to inspect your log file. Please use [our guide to find it](https://github.com/mixxxdj/mixxx/wiki/Finding%20the%20Mixxx.log%20file) and then upload it here and we'll see if we can spot anything in the log. Thank you.

**Comment by djquartz** [comment]
Thank you for responding, I'll have to get the info tonight but I will submit asap.

**Comment by djquartz** [comment]
Please see the attached log file.
[mixxx.log](https://github.com/user-attachments/files/18017117/mixxx.log)

**Comment by Swiftb0y** [comment]
Thanks. I do spot an odd line in there:
> 20:42:41.890 Debug [VinylControlProcessor 1] Xwax Vinyl control starting with a sample rate of: 0 Hz

Can you share your `mixxx.cfg`? It should reside in the same directory as the log files.

**Comment by Swiftb0y** [comment]
Thanks. It seems this has been broken for some time now for new installs and not discovered until now. Does this match your setup? Has mixxx been installed and run on the MacOS setup you mentioned previously?

As a quick and dirty workaround, you can add this entry into your mixxx.cfg manually with a text editor:
```ini
[Soundcard]
Samplerate 48000
```
The exact number depends on your audio samplerate. Its likely 41000 or 48000, but rather than guessing just look it up in `Preferences > Sound Hardware > Sample Rate`.

**Comment by ronso0** [comment]
> [Soundcard]
Samplerate 48000

Where is that preference supposed to be read?
Do you mean to verify soundconfig.xml contains `samplerate="48000"`?

**Comment by Swiftb0y** [comment]
> Where is that preference supposed to be read?

https://github.com/mixxxdj/mixxx/blob/d18764a9258f1400f7d9922df6ba7210e4c5d18d/src/vinylcontrol/vinylcontrolxwax.cpp#L139-L142

Since that is not in the user config it defaults to zero and thus "Xwax Vinyl control starting with a sample rate of: 0 Hz" is printed. 

There is some explicit code that copies the samplerate from the soundconfig to the user config:

https://github.com/mixxxdj/mixxx/blob/16a994081176624cecf36eb5e675c4582ec9ba98/src/soundio/soundmanager.cpp#L567-L572

(which has been dropped in main likely mistakenly as part of #13649). 

All in all, the Vinyl code is in terrible shape...

**Comment by djquartz** [comment]
I will give the manual entry a try to see if it fixes things.

**Comment by Swiftb0y** [comment]
Great. then the snippet I posted should work as is. You can simply append it to `mixxx.cfg`.

**Comment by djquartz** [comment]
Yes adding the sample rate fixed the issue. I also noticed I have to tap the Vinyl button a few times for it to turn on.

**Comment by Swiftb0y** [comment]
Yeah... as I said, not in the greatest of shape... I have a turntable setup at home, so I might be able to give this some love in a future version. As of right now, I'm fairly certain its broken in `main`/`2.6`...

**Comment by djquartz** [comment]
Thanks for the help

**Comment by ronso0** [comment]
>  m_pConfig->set(ConfigKey("[Soundcard]", "Samplerate"), 

For some straneg reason Eclipse didn't find this, so all clear now, thanks for the explanation.
Do you intend to fix it?

**Comment by Swiftb0y** [comment]
not sure, as I said, the code isn't in good shape. Any quick fix doesn't inspire high confidence that its not breaking anything else in the process.

**Comment by djquartz** [comment]
I noticed there is some serious sticker drift right now too.

**Comment by Swiftb0y** [comment]
yeah, depends a little on your vinyls though. Some timecode vinyls don't contain any information to correct sticker drift, other times mixxx doesn't use that info even if its there (as I said, code is in bad shape). What timecode format does the Rane One produce?

**Comment by djquartz** [comment]
I'm using Serato 7" Control Vinyl.

**Comment by djquartz** [comment]
![da41ef5c-175f-45e2-beb7-dd617bdc66f2](https://github.com/user-attachments/assets/51bd4652-cd4a-4664-8ff6-a94e832823d2)

**Comment by NotYourAverageAl** [comment]
I got bit by this bug to on w11 on main & while testing [145699](https://github.com/mixxxdj/mixxx/pull/14569). 
Adding the sample rate fixes it

> [Soundcard]
Samplerate 48000
